suppressible-error-prone/src/main/java/com/palantir/suppressibleerrorprone/SuppressingReplacement.java

```diff
@@ -0,0 +1,101 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.suppressibleerrorprone;
+
+import com.google.common.collect.Range;
+import com.google.errorprone.fixes.Replacement;
+import com.sun.source.tree.AnnotationTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.tree.EndPosTable;
+import com.sun.tools.javac.util.JCDiagnostic.DiagnosticPosition;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class SuppressingReplacement extends Replacement {
+    // We really do need to be this lazy for generating the Replacements, as error-prone immediately converts
+    // the Fix to a Replacement when a Description is given to it, and we need to defer the computation of the
+    // Replacement until a number of Descriptions have been produced, to handle multiple errors being suppressed
+    // at the same level.
+    //
+    // We *cannot* simply make this a memoized supplier. The first thing error-prone does with the Fix is to evaluate it
+    // to produce a nice error message, and we don't want to fix the number of suppression we make until we're
+    // ready to produce the Replacement after *all* the error-prone checks have been run.
+    //
+    // There is an additional issue that by the time error-prone comes around to apply the replacements, the compiler
+    // seems to change the representation of the tree for another phase - `App.Builder` becomes `App$Builder` etc and
+    // all the source code positions change. If we calculate the replacement range too late, we insert our
+    // @SuppressWarnings at the wrong location, and the indentation is miscalculated. But we can't calculate the
+    // replacement string straight away, as we might not have all the new suppressions added yet. So we have to
+    // immediately calculate the replacement range and indentation, but hold off building the final replacement
+    // string until we have all the new suppressions.
+
+    private final Range<Integer> range;
+    private final List<String> existingSuppressions;
+    private final String suffix;
+    private final Set<String> newSuppressions;
+
+    SuppressingReplacement(
+            EndPosTable endPositions,
+            Set<String> newSuppressions,
+            Optional<CharSequence> sourceCode,
+            Optional<? extends AnnotationTree> suppressWarnings,
+            Tree tree) {
+        this.newSuppressions = newSuppressions;
+        this.range = calculateRange(endPositions, suppressWarnings, tree);
```

Comment:
perhaps move the bit about calculating the range eagerly just above this line?

Reply:
done